src/proto/THORSwap.proto

```diff
@@ -0,0 +1,55 @@
+syntax = "proto3";
+
+package TW.THORSwap.Proto;
+option java_package = "wallet.core.jni.proto";
+
+import "Bitcoin.proto";
+import "Ethereum.proto";
+import "Binance.proto";
+
+// Supported blockchains
+enum Chain {
+        THOR = 0;
+        BTC = 1;
+        ETH = 2;
+        BNB = 3;
+}
+
+// Represents an asset.  Examples: BNB.BNB, RUNE.RUNE, BNB.RUNE-67C
+message Asset {
+    Chain chain = 1;
+    string symbol = 2;
+    string token_id = 3;
+}
+
+// Input for a swap between source and destination chains; for creating a TX on the source chain.
+message SwapInput {
+    // Sender address (on the source chain)
+    string from_address = 1;
+
+    // Destination chain/asset
+    Asset to_asset = 2;
+
+    // Destination address (on the destination chain)
+    string to_address = 3;
+
+    // Vault address.  Address of the vault, on the source network, must match the pair
+    string vault_address = 4;
+
+    // Amount, as string, in the native asset of the source chain
+    string amount = 5;
```

Comment:
can we specify amount of the output? if I want to swap n ETH > 100 DAI, I only want to receive specific output

Reply:
True.  This is still WIP, and I think this proto file is not yet complete, not used in the code yet.
Relevant implementation is
https://github.com/trustwallet/wallet-core/pull/1371/files/8c9ff3c3fe33de9c4e2bb2671157a29c25feb256?file-filters%5B%5D=.cpp&file-filters%5B%5D=.h#diff-c23295fdbf66c0bda14c745be87141a6df0b710b68bdccdfff2bfafacd7d2f4aR28
which has `toAmountLimit`, which is the minimum output amount we are willing to accept.  Actual amount depends on current LP exchange rate and fee levels.